scripts/us_bts/latch/process.py

```diff
@@ -0,0 +1,359 @@
+            ]:
+                prop = prop[0].lower() + prop[1:]
+                pvs.append(f"measuredProperty: dcs:{prop}")
+                sv_pvs["measuredProperty"] = f"dcs:{prop}"
+            elif prop in [
+                    "1Person", "2Person", "3Person", "4Person", "5Person"
+            ]:
+                pv = HOUSEHOLD_PV.format(person=prop.replace('Person', ''))
+                pvs.append(f"householdSize: {pv}")
+                sv_pvs["householdSize"] = f"{pv}"
+            elif "AvailableVehicles" in prop:
+                prop = prop.replace("With", "")\
+                    .replace("AvailableVehicles", "")
+                pv = NUM_OF_VEHICLES_PV.format(vehicle=prop)
+                pvs.append(f"numberOfVehicles: {pv}")
+                sv_pvs["numberOfVehicles"] = f"{pv}"
+            elif prop in ["Urban", "Rural", "SemiUrban"]:
+                pvs.append(f"placeOfResidenceClassification: dcs:{prop}")
+                sv_pvs["placeOfResidenceClassification"] = f"dcs:{prop}"
+        resolved_dcid = get_statvar_dcid(sv_pvs)
+        dcid_nodes[dcid] = resolved_dcid
+        mcf_nodes.append(
+            MCF_TEMPLATE.format(dcid=resolved_dcid, xtra_pvs='\n'.join(pvs)))
+    mcf = '\n'.join(mcf_nodes)
+    # Writing Genereated MCF to local path.
+    with open(mcf_file_path, 'w+', encoding='utf-8') as f_out:
+        f_out.write(mcf.rstrip('\n'))
+    return dcid_nodes
+
+
+def _generate_tmcf(tmcf_file_path: str) -> None:
+    """
+    This method generates TMCF file w.r.t
+    dataframe headers and defined TMCF template.
+    Args:
+        tmcf_file_path (str): Output TMCF File Path
+    """
+    with open(tmcf_file_path, 'w+', encoding='utf-8') as f_out:
+        f_out.write(TMCF_TEMPLATE.rstrip('\n'))
+
+
+def process(input_files: list, cleaned_csv_file_path: str, mcf_file_path: str,
+            tmcf_file_path: str):
+    """
+    Process Input Raw Files and apply transformations to generate final
+    CSV, MCF and TMCF files.
+    """
+    final_cols = ["year", "location", "sv", "observation", "measurement_method"]
+    final_df = pd.DataFrame()
+    sv_names = []
+    for file_path in input_files:
+        if "latch_2017-b" in file_path:
+            conf = CONF_2017_FILE
+        elif "NHTS_2009_transfer" in file_path:
+            conf = CONF_2009_FILE
+        data_df = _process_household_transportation(file_path, conf)
+        data_df = data_df.dropna(subset=["observation"])
+        final_df = pd.concat([final_df, data_df[final_cols]])
+        sv_names += data_df["sv"].to_list()
+    final_df = _promote_measurement_method(final_df)
+    sv_names = list(set(sv_names))
+    sv_names.sort()
+    updated_sv = _generate_mcf(sv_names, mcf_file_path)
+    final_df["sv"] = final_df["sv"].map(updated_sv)
+    _generate_tmcf(tmcf_file_path)
+    final_df.to_csv(cleaned_csv_file_path, index=False)
+
+
+def main(_):
+    input_path = _FLAGS.input_path
+    try:
+        ip_files = os.listdir(input_path + "abcd")
```

Comment:
Can you explain, what is abcd here?

Reply:
"abcd" is added to test logger functionality if input path is invalid. Removed it.